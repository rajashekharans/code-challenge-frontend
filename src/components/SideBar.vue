<template>
  <a-layout-sider
    breakpoint="lg"
    collapsed-width="40"
    :style="{ minWidth: '300px', height: '100vh'}"
  >
    <p id="select-property-header">
      <span class="nav-text">MapBox Vue App</span>
    </p>
    <p id="select-property-title">
      <span class="nav-text">To navigate,</span>
      <br/>
      <span class="nav-text">select property from drop down</span>
    </p>

    <a-select
      placeholder="Select the property"
      style="width: 250px"
      @change="(property) => setProperty({ property })"
    >
      <a-select-option v-for="property in properties" :key="property.id">
        {{ property.title }}
      </a-select-option>
    </a-select>

  </a-layout-sider>
</template>

<script>

import { mapGetters, mapActions } from 'vuex';

export default {
  name: 'SideBar',
  computed: {
    ...mapGetters('geoJsonSource', {
      properties: 'getAllProperties',
    }),
  },
  methods: {
    ...mapActions('geoJsonSource', [
      'setProperty',
    ]),
  },
};
</script>
<style scoped>
  #select-property-title {
    margin-top: 40px;
    color: #fff;
  }
  #select-property-header {
    margin-top: 40px;
    color: #fff;
    font-size: x-large;
  }
</style>
