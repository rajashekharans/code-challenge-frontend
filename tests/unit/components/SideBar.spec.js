import Vuex from 'vuex';
import { createLocalVue, shallowMount } from '@vue/test-utils';
import SideBar from '@/components/SideBar.vue';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('SideBar', () => {
  let store;

  beforeEach(() => {
    const geoJsonSourceState = {
      all:[],
    };

    const geoJsonSourceActions = {
      setProperty: jest.fn(),
    };

    const geoJsonSourceGetters = {
        getAllProperties: jest.fn(),
      };

    store = new Vuex.Store({
      modules: {
        geoJsonSource: {
          namespaced: true,
          state: geoJsonSourceState,
          getters: geoJsonSourceGetters,
          actions: geoJsonSourceActions,
        },
      },
    });
  });

  it('renders app heading when component loads', () => {
    const wrapper = shallowMount(SideBar, {
      store,
      localVue,
    });
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find('#select-property-header').exists()).toBe(true);
  });

  it('renders dropdown description when component loads', () => {
    const wrapper = shallowMount(SideBar, {
      store,
      localVue,
    });

    expect(wrapper.find('#select-property-title').exists()).toBe(true);
  });
});