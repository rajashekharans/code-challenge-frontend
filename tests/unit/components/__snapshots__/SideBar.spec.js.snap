// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar renders app heading when component loads 1`] = `
<a-layout-sider breakpoint="lg" collapsed-width="0" style="min-width: 300px; height: 100vh;">
  <p id="select-property-header"><span class="nav-text">MapBox Vue App</span></p>
  <p id="select-property-title"><span class="nav-text">To navigate,</span> <br> <span class="nav-text">select property from drop down</span></p>
  <a-menu theme="dark" mode="inline">
    <a-select placeholder="Select the property" style="width: 250px;"></a-select>
  </a-menu>
</a-layout-sider>
`;
